p2p/chain.py

```diff
@@ -720,6 +610,138 @@ class RegularChainSyncer(FastChainSyncer):
         return head.block_number
 
 
+class PeerRequestHandler:
+
+    def __init__(self, db: 'AsyncHeaderDB', logger: TraceLogger, token: CancelToken) -> None:
+        self.db = db
+        self.logger = logger
+        self.cancel_token = token
+
+    async def handle_get_block_bodies(self, peer: ETHPeer, block_hashes: List[Hash32]) -> None:
+        chaindb = cast('AsyncChainDB', self.db)
+        bodies = []
+        # Only serve up to eth.MAX_BODIES_FETCH items in every request.
+        for block_hash in block_hashes[:eth.MAX_BODIES_FETCH]:
+            try:
+                header = await wait_with_token(
```

Comment:
Maybe put the `self.wait(...)` API on this class.  Maybe we can introduce that as a mixin class that 

```python
class Cancelable:
    cancel_token: CancelToken = None

    def wait(...):
        ...

    def wait_first(...):
        ...
```

Or Just inline a duplication of the `wait` function from our `Service` class with a comment/issue to remove the duplication at some point.

Reply:
I like the mixin idea! Just thought I'd add a `Mixin` suffix to its name to make that clear, though